controllers/search_controller.rb

```diff
@@ -13,7 +13,46 @@ class SearchController < ApplicationController
     end
 
     get '/metadata' do
-      process_search
+      query = get_query(params)
+      options = get_ontology_metadata_search_options(params)
+      page, page_size = page_params
+
+      resp = search(Ontology, query, options)
+
+      result = {}
+      acronyms_ids = {}
+      resp.each do |doc|
+        resource_id = doc["resource_id"]
+        id = doc["submissionId_i"]
+        acronym = doc["ontology_t"].split('/').last
+        next if acronym.blank?
+        
+        already_found_submission_id = acronyms_ids[acronym]
+        old_id = already_found_submission_id.to_i rescue 0
+        already_found = (old_id && id && (id <= old_id))
+
+        next if already_found
+
+        not_restricted = (doc["ontology_viewingRestriction_t"]&.eql?('public') || current_user&.admin?)
+        user_not_restricted = not_restricted ||
+          Array(doc["ontology_viewingRestriction_txt"]).any? {|u| u.split(' ').last == current_user&.username} ||
+          Array(doc["ontology_acl_txt"]).any? {|u| u.split(' ').last == current_user&.username}
+
+        user_restricted = !user_not_restricted
+        next if user_restricted
+        
+        acronyms_ids[acronym] = id
+        result[acronym] = {
```

Comment:
the search metadata endpoint need to return SemanticArtefact

Reply:
we can do it but i think this is going to slow the search, no ?, because if we have lot of results we will have to call the SemanticArtefact to have the artefacts one by one. 
The other thing is that we have to bring also in the artefacts the fields used in the search and i think the users only want the fields used in the search.

Reply:
All the attribute are already indexed in Solr so need need to do a bring to the triple store, all you need is to transform the Solr result to a SemanticArtefact model

Reply:
i used this one to transform the results into SemanticArtefacts objects

```ruby
result[acronym] = LinkedData::Models::SemanticArtefact.read_only(id: "#{LinkedData.settings.id_url_prefix}artefacts/#{acronym}", acronym: acronym, description: doc['description_text'], title: doc['ontology_name_text'])
```

Reply:
yeah seems fine, we will check once deployed in StagePoral